manager/eris-mint/evm/snative.go

```diff
@@ -247,14 +247,27 @@ func (contract *SNativeContractDescription) Dispatch(appState AppState,
 	return function.F(appState, caller, remainingArgs, gas)
 }
 
-// We define the address of an SNative contact as the simplest possible hash of
-// its canonical name
-func (contract *SNativeContractDescription) Address() Word256 {
-	return LeftPadWord256([]byte(contract.Name))
+// We define the address of an SNative contact as the first 20 bytes of the sha3
+// hash of its name
+func (contract *SNativeContractDescription) Address() abi.Address {
+	var address abi.Address
+	copy(address[:], sha3.Sha3([]byte(contract.Name))[:abi.AddressLength])
```

Comment:
that's of a public key though, so it's kind of unrelated... We could copy, but I can't see how it will make any difference. The point is just to have a uniform hash over the 20 bytes.